app/assets/javascripts/discourse/app/lib/wait-for-keyboard.js

```diff
@@ -0,0 +1,66 @@
+import { getOwner } from "@ember/application";
+
+export async function waitForKeyboard(context) {
+  return new Promise((resolve) => {
```

Comment:
Wrapping an entire `async` function in `new Promise()` feels a bit weird. How about something like this? IMO it reads much nicer since you can follow it top-to-bottom:

```js
export async function waitForKeyboard(context){
  if (!window.visualViewport) {
    return { visible: false };
  }

  ...

  let timeout;
  let viewportListener;
  await Promise.race(
    new Promise(resolve => timeout = setTimeout(resolve, 1000)),
    new Promise(resolve => window.visualViewport.addEventListener('resize', viewportListener = resolve))
  )
  clearTimeout(timeout);
  window.visualViewport.removeEventListener('resize', viewportListener);

  // Now do the logic which is currently in `onViewportResize`
}
```

Reply:
Hmmm, that feels more convulted to me just to avoid the wrap, but if you prefer Im ok with this.

Reply:
Ok fair enough. It's your code - happy to go with your preference here :+1:

Reply:
No it's ok, I have already started the change